common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -42,28 +42,44 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     })
   }
 
-  sealed abstract class AmpExternalVideo(val videoId: String, val elementType: String)
-  case class YoutubeExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-youtube")
-  case class VimeoExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-vimeo")
+
+  sealed abstract class AmpExternalVideo(val videoId: String, val elementType: String, customAttributes: String => Map[String, String]){
+    val commonVideoAttributes = Map(("width", "5"), ("height", "3"), ("layout", "responsive"))
+    def getAttributes = customAttributes(videoId) ++ commonVideoAttributes
```

Comment:
If you decide to do my other suggestion then this would become `val attributes = ` I suppose